cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -267,7 +267,12 @@ public KafkaBrokerConfigurationBuilder withListeners(
         }
 
         // Security protocol and Control Plane Listener are configured everywhere
+
+        // Control Plane listener is added to security protocol on KRaft broker only nodes as well because validated by Kafka
+        // This is because it will be in the controller.listener.names to allow connecting to controllers
         securityProtocol.add(CONTROL_PLANE_LISTENER_NAME + ":SSL");
+        // Control Plane listener is configured on KRaft broker only nodes as well for allowing TLS certificates keystore generation
+        // so that brokers are able to connect to controllers as TLS clients
```

Comment:
Maybe this would be a bit more clear?

```suggestion
        // Brokers need to know how to connect to the controllers on the Control Plane listener and what security (encryption/authentication) they should use.
        // For that reason, we have to configure the Control Plane listener in the broker-only configuration as well, even though they do not listen at the Control Plane listener port.
        // The brokers use this configuration to detect how to connect to the controllers, what certificates to use etc.
        securityProtocol.add(CONTROL_PLANE_LISTENER_NAME + ":SSL");
```

Reply:
Just pushed this change, @fvaleri @katheris any more comments?

Reply:
Fine for me, thanks.

Reply:
Yeah that looks good to me